source/matplot/backend/backend_interface.h

```diff
@@ -213,7 +214,31 @@ namespace matplot {
             /// This is useful when tracing the gnuplot commands
             /// and when generating a gnuplot file.
             virtual void include_comment(const std::string &text);
-        };
+        }; // class backend_interface
+
+        /// Captures (backend) version information.
+        struct Version {
+            int major{0}, minor{0}, patch{0};
+            constexpr bool operator==(const Version &o) const {
+                return major == o.major && minor == o.minor && patch == o.patch;
+            }
+            constexpr bool operator!=(const Version &other) const { return !(*this == other); }
+            constexpr bool operator<(const Version &other) const {
+                if (major < other.major)
```

Comment:
Is it the same as `std::tie(major, minor, patch) < std::tie(other.major, other.minor, other.patch)`?

Reply:
Yes. Looking at the rest of the code and considering `tuple` defines `operator<`, I do prefer `Version` over `tuple` here if we were coming up with the original API now, but I don't think breaking it now brings a lot of value compared to the potential hassle of people opening issues because the API changed.

Reply:
This is gone, sorry my bad.